src/main/java/com/vinylteam/vinyl/web/controller/CatalogController.java

```diff
@@ -20,12 +21,22 @@
 public class CatalogController {
 
     private final UniqueVinylService uniqueVinylService;
+    private final DefaultCatalogService catalogService;
+
     private final UniqueVinylMapper uniqueVinylMapper;
+    private final OneVinylOfferMapper oneVinylOfferMapper;
+
+    private final int amountOfRandomVinyls = 50;
 
     @GetMapping
     public List<UniqueVinylDto> getCatalogPage() {
-        List<UniqueVinyl> uniqueVinyls = uniqueVinylService.findRandom(50);
-        return uniqueVinylMapper.uniqueVinylsToUniqueVinylDtoList(uniqueVinyls);
+        return catalogService.findRandomUniqueVinyls(amountOfRandomVinyls);
+    }
+
+    @GetMapping("/{id}")
+    public OneVinylPageDto getOneVinylOfferPage(@SessionAttribute(value = "user", required = false) User user,
```

Comment:
I think that User is unnecessary. We needed this in past when we used Cookies

Reply:
Thank you for pointing it out, I will fix it in later pr